views/settings.tpl.html

```diff
@@ -1,984 +1,1350 @@
-                               type="password" id="password_old"
-                               name="password_old" placeholder="Old password" minlength="6" required>
-                    </div>
+        <form class="w-full md:w-3/4">
+          <div class="flex mb-8">
+            <div class="inline-block w-1/2 mr-4">
+              <label class="font-semibold text-foreground" for="register-webauthn">Security Keys & Biometric
+                Authentication</label>
+              <span class="block text-sm text-muted">
+                Set up passwordless authentication using security keys, fingerprint, or face recognition for faster and
+                more secure login.
+              </span>
+            </div>
+            <div class="w-1/2 ml-4">
+              <!-- Existing Credentials List -->
+              <div id="credentials-list" class="mb-4">
+                <h4 class="mb-3 text-sm font-medium text-foreground">Your registered authenticators:</h4>
+                <div id="credentials-container">
+                  <!-- Credentials will be loaded here -->
+                  <div class="text-sm text-muted">Loading...</div>
                 </div>
+              </div>
+            </div>
+          </div>
+
+          <div class="flex mb-8">
+            <div class="inline-block w-1/2 mr-4">
+              <label class="font-semibold text-foreground" for="register-webauthn">Add a new authenticator</label>
+              <span class="block text-sm text-muted">
+                Click the button below and follow the prompts to register a new security key, fingerprint, or other
+                biometric authentication method.
+              </span>
+            </div>
+            <div class="w-1/2 ml-4">
+              <!-- WebAuthn Success/Error Messages -->
+              <div id="webauthn-error" class="mb-4 alert alert-error" style="display: none;"></div>
+              <div id="webauthn-success" class="mb-4 alert alert-success" style="display: none;"></div>
+
+              <div class="mb-4" id="credential-name-input" style="display: none;">
+                <input class="input-default" type="text" id="credential-nickname"
+                  placeholder="Enter a nickname for this authenticator (optional)" maxlength="50">
+                <p class="mt-1 text-xs text-muted">Give your authenticator a memorable name like "YubiKey" or "iPhone"
+                </p>
+              </div>
+
+              <div class="flex items-center space-x-2">
+                <button type="button" id="register-webauthn" class="btn-primary" disabled>
+                  <svg class="inline w-4 h-4 mr-2" fill="currentColor" viewBox="0 0 20 20">
+                    <path fill-rule="evenodd"
+                      d="M10 3a1 1 0 011 1v5h5a1 1 0 110 2h-5v5a1 1 0 11-2 0v-5H4a1 1 0 110-2h5V4a1 1 0 011-1z"
+                      clip-rule="evenodd" />
+                  </svg>
+                  Add Authenticator
```

Comment:
Minor: I don't really like the fact that this button has two different functions. On first click, it causes the input field to show up, on second click, it actually performs the API request. Simply display the input field by default (without button click) and give this button only a "save" functionlity.